recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -0,0 +1,210 @@
+import os
+import textwrap
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+required_conan_version = ">=1.33.0"
+
+
+class OpenTelemetryCppConan(ConanFile):
+    name = "opentelemetry-cpp"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/open-telemetry/opentelemetry-cpp"
+    description = "The C++ OpenTelemetry API and SDK"
+    requires = [
+        "abseil/20210324.2",
+        "grpc/1.37.1",
+        "libcurl/7.79.1",
+        "nlohmann_json/3.10.4",
+        "opentelemetry-proto/0.11.0",
+        "openssl/1.1.1l",
+        "protobuf/3.17.1",
+        "thrift/0.14.2"
+    ]
+    topics = ("opentelemetry", "telemetry", "tracing", "metrics", "logs")
+    generators = "cmake", "cmake_find_package_multi"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+    }
+    exports_sources = "CMakeLists.txt"
+
+    short_paths = True
+    _cmake = None
+
+    def validate(self):
+        if self.settings.arch != "x86_64":
```

Comment:
What would be needed to provided this for armv8 as well?

Reply:
Per OpenTelemetry C++ docs only x86-64 is officially supported /and CMake is not officially supported on MacOS): https://github.com/open-telemetry/opentelemetry-cpp#supported-development-platforms

I think it would need to be tested, but that would entail work in OTel C++, so perhaps it might be lifted in newer versions.